UT4MasterServer/Services/StatisticsService.cs

```diff
@@ -0,0 +1,2203 @@
+				AccountId = epicAccountId,
+				Type = StatisticType.FCKills,
+				Value = statisticBulkDto.FCKills.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.FCKillPoints.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.FCKillPoints,
+				Value = statisticBulkDto.FCKillPoints.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.FlagSupportKills.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.FlagSupportKills,
+				Value = statisticBulkDto.FlagSupportKills.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.FlagSupportKillPoints.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.FlagSupportKillPoints,
+				Value = statisticBulkDto.FlagSupportKillPoints.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.RegularKillPoints.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.RegularKillPoints,
+				Value = statisticBulkDto.RegularKillPoints.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.FlagGrabs.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.FlagGrabs,
+				Value = statisticBulkDto.FlagGrabs.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.AttackerScore.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.AttackerScore,
+				Value = statisticBulkDto.AttackerScore.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.DefenderScore.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.DefenderScore,
+				Value = statisticBulkDto.DefenderScore.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.SupporterScore.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
+				CreatedAt = currentDateTime,
+				AccountId = epicAccountId,
+				Type = StatisticType.SupporterScore,
+				Value = statisticBulkDto.SupporterScore.Value,
+				Window = StatisticWindow.Daily,
+				OwnerType = ownerType
+			});
+		}
+
+		if (statisticBulkDto.TeamKills.HasValue)
+		{
+			newStatistics.Add(new Statistic()
+			{
```

Comment:
This copy paste is alright for now, but this definitely needs some improvement in the future...

Reply:
It is possible to do it in one foreach loop using reflection, but that is way slower. I don't recommend it.

Reply:
Yea, definitely don't want to use reflection. I just thought i'd comment about it.